src/include/di/injectors.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "di/di_help.h"
+#include "storage/garbage_collector.h"
+#include "storage/record_buffer.h"
+#include "storage/storage_defs.h"
+#include "storage/write_ahead_log/log_manager.h"
+#include "transaction/transaction_manager.h"
+
+namespace terrier::di {
+/**
+ * When applied, returns an injector that contains object bindings with the correct scope for the storage
+ * engine (storage, logging, GC, transactions). Note that this does not contain bindings for constants
+ * such as object pool sizes. Unless those are supplied in the top level injector, the code will not compile.
+ */
+// TODO(Tianyu): Should we provide default values for constants here?
+auto storage_injector UNUSED_ATTRIBUTE = [] {
```

Comment:
What are the circumstances when we should define a new injector here? Any time we have a set of components that we want to bring up together?

Reply:
I think so. Whatever corresponds to a "layer" or "engine" in the system might qualify. I don't have a tried and true rule to provide.